theforeman.org/pipelines/release/pipelines/foreman-rpm.groovy

```diff
@@ -9,42 +9,114 @@ pipeline {
+                    steps {
+                        script {
+                            runDuffyPipeline('foreman-rpm', foreman_version)
+                        }
+                    }
+                }
+                stage('koji-push-rpms') {
+                    agent { label 'admin && sshkey' }
+                    steps {
+                        script {
+                            for (release in foreman_el_releases) {
+                                push_rpms_direct("foreman-${foreman_version}/${release}", "${foreman_version}/${release}")
+                            }
+                        }
+                    }
                 }
             }
         }
-        stage('Push RPMs') {
-            agent { label 'admin && sshkey' }
-            steps {
-                script {
-                    for (release in foreman_el_releases) {
-                        push_rpms_direct("foreman-${foreman_version}/${release}", "${foreman_version}/${release}")
+        stage('staging') {
+            agent { label 'el8' }
+            when {
+                expression { stage_source == 'stagingyum' }
+            }
+            stages {
+                stage('staging-build-repository') {
+                    steps {
+                        git url: "https://github.com/theforeman/theforeman-rel-eng", poll: false
+
+                        script {
+                            foreman_el_releases.each { distro ->
+                                sh "./build_stage_repository foreman ${foreman_version} ${distro}"
+                            }
+                        }
+                    }
+                }
+                stage('staging-copy-repository') {
+                    steps {
+                        script {
+                            dir('tmp') {
+                                rsync_to_yum_stage('foreman', 'foreman', foreman_version)
+                            }
+                        }
+                    }
+                }
+                stage('staging-repoclosure') {
+                    steps {
+                        script {
+                            def parallelStagesMap = [:]
+                            def name = 'foreman-staging'
+                            foreman_el_releases.each { distro ->
+                                parallelStagesMap[distro] = { repoclosure(name, distro, foreman_version) }
+                            }
+                            parallel parallelStagesMap
+                        }
+                    }
+                    post {
+                        always {
+                            deleteDir()
+                        }
+                    }
+                }
+                stage('staging-install-test') {
+                    agent any
+
+                    steps {
+                        script {
+                            runDuffyPipeline('foreman-rpm', foreman_version)
```

Comment:
should/could this pass `staging_source` somehow, so we don't need https://github.com/theforeman/forklift/pull/1717?

Reply:
mhh, `runDuffyPipeline` doesn't allow us to pass arbitrary parameters to the job, so probably not so easy.

Reply:
I reached the same conclusion. However, forklift should match CI if someone goes to run things so I think it's OK.